gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/javaversions/BaselineJavaVersions.java

```diff
@@ -111,24 +111,21 @@ private static IsLibraryWithReason isLibrary(Project project, BaselineJavaVersio
         }
 
         PublishingExtension publishing = project.getExtensions().findByType(PublishingExtension.class);
-        if (publishing == null) {
-            return new IsLibraryWithReason(false, "doesn't have any publishing extensions defined");
+        if (publishing != null) {
+            return new IsLibraryWithReason(
+                    true,
+                    String.format(
+                            "has publishing extensions with publications %s",
+                            publishing.getPublications().getNames()));
```

Comment:
Based on the previous code, this seems to be an edge case for when _something_ is publishing, even though it's not using one of our library publishing plugins, so I'd keep some of the mentions regarding this
```suggestion
        // If the project seems to publish something without our publishing plugins,
        //   be conservative and consider this a library, to avoid releasing a library with too high a java version
        if (publishing != null) {
            return new IsLibraryWithReason(
                    true,
                    String.format(
                            "has publishing extensions with publications %s.\n" +
                            "Despite not having any library publish plugins (%s), this is conservatively regarded as a library for safety.",
                            publishing.getPublications().getNames(),
                            LIBRARY_PLUGINS));
```

Reply:
I'm not sure this is particularly relevant. It seems simpler to just say "anything with a publishing extension is considered a library".

Reply:
I defer to @felixdesouza on what he'd prefer here